cmd/kubectl-kubestellar/cmd/root.go

```diff
@@ -0,0 +1,117 @@
+// key is set to "kubectl kubestellar", then "kubestellar" gets dropped from
+// the usage command string. This is an open issue in Cobra for kubectl plugins.
+// The workaround is to add "kubectl" along with a non-breaking space to the
+// "Use" string, but this will break the autocompletion script generator.
+// Luckily, the help printed to the screen does contain a regular space that
+// can be copied and pasted without issue.
+// Having accurate help is more important than autocompletion, so we will ensure
+// help prints the full command string, and we will disable the "completion"
+// script generator command with a dummy function.
+// See https://github.com/spf13/cobra/issues/2017
+var rootCmd = &cobra.Command{
+	Use:   "kubectl\u00A0kubestellar",
+	Short: "KubeStellar plugin for kubectl",
+	Long: `KubeStellar is a flexible solution for challenges associated with multi-cluster 
+configuration management for edge, multi-cloud, and hybrid cloud.
+This command provides the kubestellar sub-command for kubectl.`,
+	Args: cobra.ExactArgs(1),
+	// If an invalid sub-command is sent, the function in RunE will execute.
+	// Use this to inform of invalid arguments, and return an error.
+	// For details on inconsistent error behavior, see
+	// https://github.com/spf13/cobra/issues/706
+	RunE: func(cmd *cobra.Command, args []string) error {
+		if len(args) > 0 {
+			// NOTE this only runs if "Args:  cobra.ExactArgs(1)" is set; if not
+			// set the error message is brief and does not print the help.
+			return errors.New(fmt.Sprintf("Invalid kubestellar sub-command: %s\n", args[0]))
+		} else {
+			// NOTE this does not run if "Args:  cobra.ExactArgs(1)" is set,
+			// although the error message printed is acceptable.
+			return errors.New("Missing kubestellar sub-command\n")
+		}
+	},
+}
+
+// Dummy function to disable auto-completion script generation, since this
+// feature is broken (see comments above rootCmd).
+var completionCmd = &cobra.Command{
+	Use:   "completion",
+	Short: "Generate the autocompletion script for the specified shell",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		return errors.New("Not implemented\n")
+	},
+}
+
+func init() {
+	// Get config flags with default values.
+	// Passing "true" will "use persistent client config, rest mapper,
+	// discovery client, and propagate them to the places that need them,
+	// rather than instantiating them multiple times."
+	cliOpts := genericclioptions.NewConfigFlags(true)
+	// Make a new flag set named en
+	fs := pflag.NewFlagSet("root", pflag.ExitOnError)
+	// Add cliOpts flags to fs (flow from syntax is confusing, goes -->)
+	cliOpts.AddFlags(fs)
+
+	// Allow logging to pick up flags
+	klog.InitFlags(flag.CommandLine)
+	// Add logging flags to fs
+	fs.AddGoFlagSet(flag.CommandLine)
+	// Add flags to our command; make these persistent (available to this
+	// command and all sub-commands)
+	rootCmd.PersistentFlags().AddFlagSet(fs)
```

Comment:
You don't want two different `cliOpts` adding the same flags to the persistent flag set. If you add the root command's `cliOpts`'s flags then do not also add the flags of different `cliOpts` created for subsidiary commands.

Remember that `kubectl kubestellar deploy` is different from all the others, it does NOT want the flags of a `cliOpts`, it wants the flags for `helm create` / `helm apply`.